nav2_smac_planner/src/a_star.cpp

```diff
@@ -211,8 +253,11 @@ void AStarAlgorithm<NodeT>::setGoal(
       _collision_checker->getCostmapROS(), _start->pose.x, _start->pose.y, mx, my);
   }
 
-  _goal_coordinates = goal_coords;
-  _goal->setPose(_goal_coordinates);
+  _goals_coordinates = goals_coordinates;
```

Comment:
You are right about the _goal, yes, but the _goal_coordinates i would have to mantain in order to check that a start has been set but this would require i not clear the _goal_coordinate initially.